hr_course/models/hr_course.py

```diff
@@ -180,6 +183,41 @@ def cancel_course(self):
         for record in self:
             record.write(record._cancel_course_values())
 
+    @api.model
+    def send_course_notification_email(self, partner_emails):
```

Comment:
to be rewritten with a mail.channel. Since hr.course is a mail.thread, you can use the message_post_with_template method with the channel_id argument to give the channel.

Reply:
@gurneyalex  Updated, but I didn't find the usage of message_post_with_template with channel_id as a param, so i am not very sure about it.

Couple of questions about using mail.channel from my side:

1) in data from many other modules i see defined channel and `mail.message` assigned to it, for example:

```
<!-- Discussion groups, done in 2 steps to remove creator from followers -->
        <record model="mail.channel" id="channel_1">
            <field name="name">sales</field>
            <field name="description">Discussion about best sales practices and deals.</field>
        </record>
        .....
        <!-- Best sales practices messages -->
        <record id="mail_message_channel_1_1" model="mail.message">
            <field name="model">mail.channel</field>
            <field name="res_id" ref="mail.channel_1"/>
            <field name="body"><![CDATA[<p>Selling a training session and selling the products after the training session is more efficient than directly selling a pack with the training session and the products.</p>]]></field>
            <field name="message_type">comment</field>
            <field name="subtype_id" ref="mt_comment"/>
            <field name="author_id" ref="base.partner_demo"/>
            <field name="date" eval="(DateTime.today() - timedelta(days=5)).strftime('%Y-%m-%d %H:%M')"/>
        </record>
```

should I use mail.message instead of mail.template defined in data?

2) in mail.channel configuration i see members assigned, I guess here we need the same members as in the course? I'd say we could update channel with users at the moment of assigning employees to the course...but what if many courses with different employees enrolled are using the same channel? Or we need separate channel per course?

3) I can see field for mail alias on mail.channel - how to configure it properly? I guess putting random string isn't enough to work.

afaiu mail channel is only collecting data about message (mail alias, groups, recipients etc), but referring to the code at the beginning of my comment, I can't find out how do we use this message and mail channel to send anything.